code/TagField.php

```diff
@@ -8,34 +8,142 @@
  */
 class TagField extends DropdownField {
 	/**
+	 * @var array
+	 */
+	public static $allowed_actions = array(
+		'suggest',
+	);
+
+	/**
 	 * @var bool
 	 */
-	protected $readOnly;
+	protected $ajax = false;
+
+	/**
+	 * @var bool
+	 */
+	protected $readOnly = false;
+
+	/**
+	 * @var string
+	 */
+	protected $relationTitle = 'Title';
+
+	/**
+	 * @var int
+	 */
+	protected $limit = 10;
 
 	/**
 	 * @var null|string
 	 */
-	protected $relationTitleField;
+	protected $recordClass;
 
 	/**
-	 * @param string      $name
+	 * @param string $name
 	 * @param null|string $title
-	 * @param array       $source
-	 * @param array       $value
-	 * @param bool        $readOnly
-	 * @param string      $relationTitleField
+	 * @param array $source
+	 * @param array $value
+	 * @param bool $readOnly
 	 */
-	public function __construct($name, $title = null, $source = array(), $value = array(), $readOnly = false, $relationTitleField = 'Title') {
+	public function __construct($name, $title = null, $source = array(), $value = array(), $readOnly = false) {
 		$this->readOnly = $readOnly;
-		$this->relationTitleField = $relationTitleField;
 
 		parent::__construct($name, $title, $source, $value);
 	}
 
 	/**
-	 * @param array $properties
+	 * @return bool
+	 */
+	public function getAjax() {
+		return $this->ajax;
+	}
+
+	/**
+	 * @param bool $ajax
 	 *
-	 * @return string
+	 * @return static
+	 */
+	public function setAjax($ajax) {
+		$this->ajax = $ajax;
+
+		return $this;
+	}
+
+	/**
+	 * @return bool
+	 */
+	public function getReadOnly() {
+		return $this->readOnly;
+	}
+
+	/**
+	 * @param bool $readOnly
+	 *
+	 * @return static
+	 */
+	public function setReadOnly($readOnly) {
+		$this->readOnly = $readOnly;
+
+		return $this;
+	}
+
+	/**
+	 * @return null|string
+	 */
+	public function getRelationTitle() {
+		return $this->relationTitle;
+	}
+
+	/**
+	 * @param string $relationTitle
+	 *
+	 * @return static
+	 */
+	public function setRelationTitle($relationTitle) {
+		$this->relationTitle = $relationTitle;
+
+		return $this;
+	}
+
+	/**
+	 * @return int
+	 */
+	public function getLimit() {
```

Comment:
Unclear if this is the limit for auto-suggest, the allowed number of tags to be stored on a relation, or the number of tags loaded from an existing relation. Needs PHPDoc.